src/Horse.CGI.pas

```diff
@@ -0,0 +1,74 @@
+unit Horse.CGI;
+
+interface
+
+uses IdHTTPWebBrokerBridge, Horse.API, IdContext;
+
+type
+  THorse = class(THorseAPI)
+  private
+    FPort: Integer;
+    procedure OnAuthentication(AContext: TIdContext; const AAuthType, AAuthData: String; var VUsername, VPassword: String;
+      var VHandled: Boolean);
+  public
+    constructor Create; overload;
+    constructor Create(APort: Integer); overload;
+    property Port: Integer read FPort write FPort;
+    procedure Start; override;
+  end;
+
+implementation
+
+{ THorse }
+
+uses Horse.Constants, Horse.WebModule, System.SysUtils, Web.WebReq;
+
+constructor THorse.Create(APort: Integer);
+begin
+  inherited Create;
+  FPort := APort;
+end;
+
+constructor THorse.Create;
+begin
+  inherited Create;
+  FPort := DEFAULT_PORT;
+end;
+
+procedure THorse.OnAuthentication(AContext: TIdContext; const AAuthType, AAuthData: String; var VUsername, VPassword: String;
```

Comment:
Test if it is necessary.

Reply:
![image](https://user-images.githubusercontent.com/16382981/64308275-114ba380-cf6f-11e9-9117-d90f66c8815b.png)